packages/console/src/pages/ApplicationDetails/index.tsx

```diff
@@ -24,6 +26,35 @@ type OidcConfig = {
   token_endpoint: string;
 };
 
+type FormData = Omit<Application, 'oidcClientMetadata'> & {
+  oidcClientMetadata: {
+    redirectUris: Array<{ uri: string }>;
+    postLogoutRedirectUris: Array<{ uri: string }>;
+  };
+};
+
+const mapApplicationDataToFormData = (data: Application) => {
+  const {
+    oidcClientMetadata: { redirectUris, postLogoutRedirectUris },
+  } = data;
+
+  const convertedRedirectUri =
+    redirectUris.length > 0 ? redirectUris.map((uri) => ({ uri })) : [{ uri: '' }];
+
+  const convertedPostLogoutRedirectUri =
+    postLogoutRedirectUris.length > 0
+      ? postLogoutRedirectUris.map((uri) => ({ uri }))
+      : [{ uri: '' }];
+
+  return {
+    ...data,
+    oidcClientMetadata: {
+      redirectUris: convertedRedirectUri,
+      postLogoutRedirectUris: convertedPostLogoutRedirectUri,
+    },
+  };
+};
```

Comment:
should we move it the api layer or data layer?

Reply:
removed